verifier/plonk/gates/mul_extension_gate.go

```diff
@@ -0,0 +1,66 @@
+package gates
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/Electron-Labs/plonky2-groth16-verifier/goldilocks"
+	algebra "github.com/Electron-Labs/plonky2-groth16-verifier/goldilocks/extension"
+	"github.com/consensys/gnark/frontend"
+)
+
+type MulExtensionGate struct {
+	NumOps int `json:"num_ops"`
+}
+
+func NewMulExtensionGate(id string) *MulExtensionGate {
+	id = strings.TrimPrefix(id, "MulExtensionGate")
+	id = strings.Replace(id, "num_ops", "\"num_ops\"", 1)
+	var gate MulExtensionGate
+	err := json.Unmarshal([]byte(id), &gate)
+	if err != nil {
+		panic(fmt.Sprintln("Invalid gate id: ", id, err))
+	}
+	return &gate
+}
+
+func (gate *MulExtensionGate) EvalUnfiltered(api frontend.API, rangeChecker frontend.Rangechecker, vars EvaluationVars) []goldilocks.GoldilocksExtension2Variable {
+	const0 := goldilocks.GetVariableArray(vars.LocalConstants[0])
+	constraints := make([]goldilocks.GoldilocksExtension2Variable, gate.NumOps*D)
+
+	for i := 0; i < gate.NumOps; i++ {
+		multiplicand0 := GetLocalExtAlgebra(vars.LocalWires, gate.wires_ithMultiplicand0(i))
+		multiplicand1 := GetLocalExtAlgebra(vars.LocalWires, gate.wires_ithMultiplicand1(i))
+		output := GetLocalExtAlgebra(vars.LocalWires, gate.wires_ithOutput(i))
+		computedOutputNoReduce := algebra.ScalarMulNoReduce(api, algebra.MulNoReduce(api, multiplicand0, multiplicand1), const0)
+		// assumming computedOutputNoReduce is always greator than output
```

Comment:
I think we can add `modulus` to `computedOutputNoReduce` to mitigate this assumption. Can do the same for every gate where we have this assumption.